src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +36,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+            acceleration.y = 0.5f;
+            acceleration.z = 0.0f;
+
+            func_800286CC(globalCtx, &position, &velocity, &acceleration, ((Math_Rand_ZeroOne() * 16) + 80),
+                          ((Math_Rand_ZeroOne() * 30) + 80));
+        }
+        i++;
+    }
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static s16 D_808B85E4[] = { 0xF060, 0x0320, 0x0FA0 };
+
+void BgSpot18Basket_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+    Actor* actor = &this->dyna.actor;
+    s32 temp = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK3);
+    func_808B7710(this, globalCtx);
+    DynaPolyInfo_Alloc(&D_06002154, &temp);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, actor, temp);
+
+    Actor_ProcessInitChain(actor, sInitChain);
+    ActorShape_Init(&actor->shape, 0.0f, ActorShadow_DrawFunc_Circle, 15.0f);
+    actor->initPosRot.pos.y = actor->initPosRot.pos.y + 0.01f;
+    actor->posRot.pos.y = actor->initPosRot.pos.y;
+
+    if (Flags_GetSwitch(globalCtx, (actor->params >> 8) & 0x3F)) {
+        func_808B7BB0(this);
+        return;
+    }
+
+    func_808B7AEC(this);
+    Actor_SpawnAsChild(&globalCtx->actorCtx, actor, globalCtx, ACTOR_BG_SPOT18_FUTA, actor->posRot.pos.x, actor->posRot.pos.y,
+                       actor->posRot.pos.z, actor->shape.rot.x, actor->shape.rot.y + 0x1555, actor->shape.rot.z, -1);
+
+    if (actor->child == NULL) {
+        osSyncPrintf("\x1b[31m");
+        osSyncPrintf("Ｅｒｒｏｒ : 変化壷蓋発生失敗(%s %d)\n", "../z_bg_spot18_basket.c", 351);
+        osSyncPrintf("\x1b[m");
+        Actor_Kill(actor);
+    }
+}
+
+void BgSpot18Basket_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyJntSph(globalCtx, &this->colliderJntSph);
+}
+
+void func_808B7AEC(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7AFC;
+}
+
+void func_808B7AFC(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F)) {
+        func_800800F8(globalCtx, 4220, 80, this, 0);
+        func_808B7B58(this);
+    }
+}
+
+void func_808B7B58(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7B6C;
+    this->unk_216 = 0;
+}
+
+void func_808B7B6C(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    if (this->unk_216 >= 21) {
+        func_808B7BB0(this);
+        this->dyna.actor.child->parent = NULL;
+        this->dyna.actor.child = NULL;
+    }
+}
+
+void func_808B7BB0(BgSpot18Basket* this) {
+    this->unk_20C = 0;
+    this->actionFunc = func_808B7BCC;
+    this->unk_210 = this->unk_20C;
```

Comment:
The lines might have to be swapped but it should work like this:
```suggestion
    this->actionFunc = func_808B7BCC;
    this->unk_210 = this->unk_20C = 0;
```